### ISSUE TITLE
Feature: Cloud Provider support IAM+Instance login

### ISSUE THREAD

**Issue by mengesb** [hashicorp/terraform-provider-vault#1077]
Commensurate with #1060 and #1066, I would like to request that the following would be supported:

- AWS IAM authentication (done - #1060)
- AWS EC2 authentication (#1060 appears to be limited only to AWS IAM auth, the AWS Auth backend also supports EC2, though unsure how widely used this might be, or it's future)
- Google IAM authentication (Similar to #1060, but for Google)
- Google GCE authentication (Similar to AWS EC2 auth, Google has GCE auth - instance authentication)

### Terraform Version
1.0.0

### Affected Resource(s)
Please list the resources as a list, for example:
- vault provider is affected

### Terraform Configuration Files
Proposed from #1060:
```hcl
provider "vault" {
  address = "http://127.0.0.1:8200"
  auth_login {
    path = "auth/aws/login"
    method = "aws"
    parameters = {
      role = "dev-role-iam"
    }
  }
}
```

So would like to see something similar for Google IAM:
```hcl
provider "vault" {
  address = "http://127.0.0.1:8200"
  auth_login {
    path = "auth/gcp/login"
    method = "gcp"
    parameters = {
      role = "dev-role-iam"
    }
  }
}
```

The EC2 or GCE "instance" auth types might be a little trickier. At minimum I would like to see support for the IAM methods for both clouds. This should be comensurate with current docs:

- https://www.vaultproject.io/docs/auth/gcp
- https://www.vaultproject.io/docs/auth/aws

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
Ability to allow the Terraform Vault provider to authenticate using a cloud provider

### Actual Behavior
AWS IAM supported via #1060 pending release, GCP unsupported. TBD support for EC2/GCE methods for cloud providers

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
N/A

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1060

**Comment by nicoche** [comment]
Hi!

Any update on this Google IAM authentication? I'm happy to provide a PR.

**Comment by benashz** [comment]
Fixed in #1607 and #1618
